src/libply/aux/syscall.c

```diff
@@ -59,6 +57,16 @@ int bpf_map_create(enum bpf_map_type type, int key_sz, int val_sz, int entries)
 	return syscall(__NR_bpf, BPF_MAP_CREATE, &attr, sizeof(attr));
 }
 
+int bpf_prog_test_run(int prog_fd)
+{
+	size_t attr_sz = offsetofend(union bpf_attr, test);
+	union bpf_attr attr;
+
+	memset(&attr, 0, attr_sz);
+	attr.test.prog_fd = prog_fd;
+
+	return syscall(__NR_bpf, BPF_PROG_TEST_RUN, &attr, attr_sz);
```

Comment:
The last argument here should just be `sizeof(attr)`. Its only purpose is to let the kernel work with older binaries that were linked to old headers where new members didn't exist yet.

Reply:
I see, that would be cool. Fixed in [93bbc1b](https://github.com/iovisor/ply/pull/82/commits/93bbc1b022b4c2587e0b32aa3fd52bb752da5e2a) .